projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/customer-ticketing.ts

```diff
@@ -149,3 +149,77 @@ export function verifyTicketDoesNotExist(ticketDetails: TestTicketDetails) {
 export function visitApparelUKTicketListingPage(){
   visitPage('apparel-uk-spa/en/GBP/my-account/support-tickets', 'apparelTicketListingPage');
 }
+
+export function getTicketDetailsFromTicketListView(){
+  let ticketDetails = [];
+  const row = cy.get('cx-customer-ticketing-list').find('tbody').get('tr').eq(1);
+  row.get('td').eq(0).invoke('text').then((x) => ticketDetails.push(x));
+  row.get('td').eq(1).invoke('text').then((x) => ticketDetails.push(x));
+  row.get('td').eq(5).invoke('text').then((x) => ticketDetails.push(x));
+
+   return ticketDetails;
+}
+
+export function verifyTicketDetails(ticketDetails){
+  //assert title when available
+  cy.get(':nth-child(1) > cx-card > .cx-card > .card-body > .cx-card-container > .cx-card-label-container > :nth-child(1) > .cx-card-label').contains(ticketDetails[1]);
+  cy.get(':nth-child(4) > cx-card > .cx-card > .card-body > .cx-card-container > .cx-card-label-container > :nth-child(1) > .cx-card-label').contains(ticketDetails[2]);
+}
+
+export function createNewTicket(){
+  const testTicketDetails: TestTicketDetails = {
+    subject: 'Entering a subject',
+    message: 'Typing a message',
+    category: TestCategory.complaint,
+  };
+
+  loginRegisteredUser();
+  visitElectronicSupportTicketPage();
+  openCreateTicketModal();
+  fillTicketDetails(testTicketDetails);
+  clickSubmitAndVerifyRequestCompleted();
+  verifyCreatedTicketDetails(testTicketDetails);
+}
+
+export function clickCloseRequestButton(){
+  cy.get(':nth-child(4) > cx-card > .cx-card > .card-body > .cx-card-container > .cx-card-label-container > :nth-child(1) > .cx-card-label').contains("Open");
+  cy.get('.btn.btn-block.btn-action"').click();
+  cy.get('.cx-customer-ticket-form-body').should('exist');
+  cy.get('label > .form-control').last().type("Closing Ticket");
+  cy.get('.cx-customer-ticket-form-footer > .btn-primary').click();
+}
+
+export function verifyStatusOnListingPage(status){
+  const row = cy.get('cx-customer-ticketing-list').find('tbody').get('tr').eq(1);
+  row.get('td').eq(5).contains(status);
+}
+
+export function verifyMessageBoxExists(doesExist){
+  if(doesExist){
+    cy.get('.form-control').should('exist');
+  } else {
+    cy.get('.form-control').should('not.exist');
+  }
+}
+
+export function clickReopenRequestButton(){
+  cy.get(':nth-child(4) > cx-card > .cx-card > .card-body > .cx-card-container > .cx-card-label-container > :nth-child(1) > .cx-card-label').contains("Closed");
+  cy.get('.btn.btn-block.btn-action"').click();
+  cy.get('.cx-customer-ticket-form-body').should('exist');
+  cy.get('label > .form-control').last().type("Reopening Ticket");
+  cy.get('.cx-customer-ticket-form-footer > .btn-primary').click();
+}
+
+export function verifyMessagesAreBeingPopulated(){
```

Comment:
What kind of message?

Reply:
The interaction between customer and agent. But there since there is a chance the customer has just created ticket which means only message will be there so i just created a assert on customers side of the tab